src/drivers/imu/analog_devices/adis16507/ADIS16507.cpp

```diff
@@ -172,7 +172,9 @@ void ADIS16507::RunImpl()
 			const uint16_t DIAG_STAT = RegisterRead(Register::DIAG_STAT);
 
 			if (DIAG_STAT != 0) {
-				PX4_ERR("DIAG_STAT: %#X", DIAG_STAT);
+				PX4_ERR("self test failed, resetting. DIAG_STAT: %#X", DIAG_STAT);
+				_state = STATE::RESET;
+				ScheduleDelayed(100_ms);
```

Comment:
Maybe this should be much longer so that an actually failing sensor won't spam as aggressively? 

```suggestion
				ScheduleDelayed(3_s);
```